src/Cli/dotnet/CommonOptions.cs

```diff
@@ -277,6 +279,18 @@ internal static CliArgument<T> AddCompletions<T>(this CliArgument<T> argument, F
             argument.CompletionSources.Add(completionSource);
             return argument;
         }
+
+        internal static DynamicOption<T> AddCompletions<T>(this DynamicOption<T> option, Func<CompletionContext, IEnumerable<CompletionItem>> completionSource)
```

Comment:
Can this take an IDynamicOption?

Reply:
IMO no because then you'd lose the knowledge of the generic `T` for the DynamicOption. DynamicOption<T> is a .NET CLI concept, whereas IDynamicOption is a concept of the completions library I added.